File: solutions/training-java/src/main/java/org/example/Collections/Collections.java
package org.example.Collections;
import java.util.*;

public class Collections {

    public static class Person{
        public String name;
        public int age;
        public int employeeNo;

        public Person(String name, int age, int employeeNo) {
            this.name = name;
            this.age = age;
            this.employeeNo = employeeNo;
        }

        @Override
        public String toString() {
            return "Person{" +
                    "name='" + name + '\'' +
                    ", age=" + age +
                    ", employeeNo=" + employeeNo +
                    '}';
        }
    }

    public static void main(String[] args) {
        //12.1
        /*ArrayList<String> people = new ArrayList<>();

        people.add("Pedro");
        people.add("Pablo");
        people.add("Mario");
        people.add("Pavel");
        people.add("Marta");

        Iterator itr = people.iterator();

        while (itr.hasNext()){
            System.out.println(itr.next());
        }

        ArrayList<Person> persons = new ArrayList<>();

        Person obj;

        while (itr.hasNext()){

            obj = new Person((String) itr.next(), 23, 123213);
            persons.add(obj);
        }

        itr = persons.iterator();

        while (itr.hasNext()){
            System.out.println(itr.next().toString());
        }*/

        //12.2
        /*HashMap<Integer, String> bookDir = new HashMap<Integer, String>();

        bookDir.put(1,"It follows");
        bookDir.put(2,"HP and the philosophal stone");
        bookDir.put(3,"Run");
        bookDir.put(4,"MistBorn");
        bookDir.put(5,"Science for dummies");

        for(Map.Entry<Integer, String> book: bookDir.entrySet()){
            System.out.println(book.getKey() + "--> " + book.getValue());
        }*/

        //12.3
        TreeSet<String> set = new TreeSet<>();

        set.add("Welcome");
        set.add("to");
        set.add("The");
        set.add("Internet");
        set.add("Welcome");

        Iterator itr = set.iterator();

        while (itr.hasNext()){
            System.out.println(itr.next());
        }
    }
}

File: solutions/training-kotlin/src/main/kotlin/Collections.kt
class PersonCollection(name: String, age: Int, employeeNo: Int){
    var name = name
    var age = age
    var employeeNo = employeeNo

    override fun toString(): String {
        return "PersonData(name='$name', age=$age, employeeNo=$employeeNo)"
    }
}

fun main() {
    // 12.1

    var people = listOf("Pedro", "Mario", "Pablo", "Juan", "Maria")

    people.forEach { println(it) }

    var persons = listOf(
        PersonCollection("Pedro", 23, 2312313),
        PersonCollection("Mario", 20, 1232345),
        PersonCollection("Pablo", 25, 8462786),
        PersonCollection("Maria", 18, 7987978),
        PersonCollection("Luis", 32, 2343214),
    )

    persons.forEach { println(it.toString()) }

    // 12.2

    var map = hashMapOf<Int, String>()
    map.put(1, "It follows")
    map.put(2, "HP and the philosophal stone")
    map.put(3, "Run")
    map.put(4, "MistBorn")
    map.put(5, "Science for dummies")

    map.forEach { println("${it.key} --> ${it.value}") }

    // 12.3
    var set = setOf("Welcome", "To", "The", "Internet", "Welcome")
    set.forEach { println(it) }
}
